src/helpers/client/safari-client.ts

```diff
@@ -65,11 +68,26 @@ export class SafariClient implements Client {
       this.safariExtensionExecutors.set(request.id, { resolve, reject });
     });
 
-    this.dispatchSafariExtensionEvent(SafariExtensionEventType.Request, request);
+    this.dispatchEvent(SafariExtensionEventType.Request, request);
 
     return promise;
   };
 
+  /**
+   * Monitors transfer activity.
+   * @returns A Promise that resolves with the response from the extension.
+   */
+  public monitorTransferActivity(): Promise<any> {
```

Comment:
Should this be Promise<void> if it does not return anything useful?  Or if it does return something can you define the type?

Reply:
It does return a `Promise`, but it is resolved or rejected later when the extension dispatch the response event. I could define the type, which should be `JSONRPCResponse`.

Reply:
Ok. Ideally avoid using any and specify the type or or if nothing useful returned use unknown or something.

Reply:
Yeah, but unfortunately in our case we don't have a fixed type or object for the responses. After checking the code again I see the value we pass to resolve the promise is `JSONRPCResponse.result` which is dynamic depending on the JSONRPC call, I think we should leave it as `any` at least for now.

Reply:
Is it used by consumers or others? I’m just confused what the use case is if generic response.

Reply:
That's private, it's only being used inside the "SafariClient" to handle the extension responses from the SDK side

Reply:
Ok. In that case we should use unknown?  If its response is never used outside of our own code. Since we know it’s not used and we should avoid any

Reply:
Sure, since we are not really consuming the value I will change it to `unknown`